esp-hal/src/soc/esp32/psram.rs

```diff
@@ -12,50 +12,75 @@
 //!
 //! The `PSRAM` module is accessed through a virtual address, defined as
 //! `PSRAM_VADDR`. The starting virtual address for the PSRAM module is
-//! 0x3F800000. The `PSRAM` module size depends on the configuration specified
-//! during the compilation process. The available `PSRAM` sizes are `2MB`,
-//! `4MB`, and `8MB`.
-//!
-//! NOTE: If you want to use `PSRAM` on `ESP32` or `ESP32-S3`, it'll work only
-//! in `release` mode.
+//! 0x3F800000.
 
 /// The starting virtual address of the PSRAM (Pseudo-SRAM) region in memory.
 pub const PSRAM_VADDR_START: usize = 0x3F800000;
 
+/// The total size of the PSRAM (Pseudo-SRAM) in bytes.
+static mut PSRAM_BYTES: u32 = 0;
+
 /// Retrieves the starting virtual address of the PSRAM (Pseudo-SRAM) region in
 /// memory.
 pub fn psram_vaddr_start() -> usize {
     PSRAM_VADDR_START
 }
 
-cfg_if::cfg_if! {
-    if #[cfg(feature = "psram-2m")] {
-        const PSRAM_SIZE: u32 = 2;
-    } else if #[cfg(feature = "psram-4m")] {
-        const PSRAM_SIZE: u32 = 4;
-    } else if #[cfg(feature = "psram-8m")] {
-        const PSRAM_SIZE: u32 = 8;
-    } else {
-        const PSRAM_SIZE: u32 = 0;
-    }
+/// Returns the size of mapped PSRAM
+pub fn psram_size() -> usize {
```

Comment:
I think this is okay to use internally for the purposes of range checking in soc, but I don't think it should be `pub`.

Reply:
we can certainly return it from init (also the start address) - then we can get rid of the annoying statics